cmake/KokkosConfig.cmake.in

```diff
@@ -17,56 +18,121 @@ include(CMakeFindDependencyMacro)
 get_filename_component(Kokkos_CMAKE_DIR "${CMAKE_CURRENT_LIST_FILE}" PATH)
 include("${Kokkos_CMAKE_DIR}/KokkosTargets.cmake")
 include("${Kokkos_CMAKE_DIR}/KokkosConfigCommon.cmake")
+set(Kokkos_FOUND_INSTALL_DIR ${Kokkos_CMAKE_DIR}/../../..)
 unset(Kokkos_CMAKE_DIR)
 
-# check for conflicts
-if("launch_compiler" IN_LIST Kokkos_FIND_COMPONENTS AND "separable_compilation" IN_LIST Kokkos_FIND_COMPONENTS)
-  message(STATUS "'launch_compiler' implies global redirection of targets depending on Kokkos to appropriate compiler.")
-  message(
-    STATUS
-      "'separable_compilation' implies explicitly defining where redirection occurs via 'kokkos_compilation(PROJECT|TARGET|SOURCE|DIRECTORY ...)'"
-  )
-  message(FATAL_ERROR "Conflicting COMPONENTS: 'launch_compiler' and 'separable_compilation'")
+# components deprecation message
+if(NOT Kokkos_ENABLE_DEPRECATED_BUILD_STRATEGY)
```

Comment:
I'd be happier turning this into a proper option with default `ON` to make it easier for people to try after merging without changing default behavior. Once we have gathered more experience we can change the default to `OFF`.

Reply:
I am happy to swap the default to the old way of building. Do you think we should keep the deprecation messages in that case?

Reply:
done in https://github.com/kokkos/kokkos/pull/7582/commits/929af8d6a03115791845258b3ad179613d0b28e4